components/director/internal/domain/formation/resolver.go

```diff
@@ -490,6 +498,32 @@ func (r *Resolver) ResynchronizeFormationNotifications(ctx context.Context, form
 	return r.conv.ToGraphQL(updatedFormation)
 }
 
+func (r *Resolver) deleteSelfFormationAssignment(ctx context.Context, tnt, formationName, objectID string) error {
+	selfFATx, err := r.transact.Begin()
+	if err != nil {
+		return err
+	}
+	selfFATransactionCtx := persistence.SaveToContext(ctx, selfFATx)
+	defer r.transact.RollbackUnlessCommitted(selfFATransactionCtx, selfFATx)
+
+	formationFromDB, err := r.service.GetFormationByName(selfFATransactionCtx, formationName, tnt)
+	if err != nil {
+		log.C(ctx).Errorf("An error occurred while getting formation by name: %q: %v", formationName, err)
+		return errors.Wrapf(err, "An error occurred while getting formation by name: %q", formationName)
+	}
+
+	fa, err := r.formationAssignmentSvc.GetReverseBySourceAndTarget(selfFATransactionCtx, formationFromDB.ID, objectID, objectID)
```

Comment:
**_minor:_** This works because the source and target are the same but I wonder if it would be more readable to use `r.formationAssignmentSvc.GetBySourceAndTarget` when we get "self-referenced FA"?

Reply:
I think so too, but there is no such method in the service for formation assignments and it's already overloaded with methods.